src/initramfs/cmd/init/pkg/platform/baremetal/baremetal.go

```diff
@@ -74,3 +74,36 @@ func (b *BareMetal) UserData() (data userdata.UserData, err error) {
 func (b *BareMetal) Prepare(data userdata.UserData) (err error) {
 	return nil
 }
+
+func (b *BareMetal) Install(data userdata.UserData) (err error) {
+	// No installation necessary
+	if data.Install == nil {
+		return nil
+	}
+
+	if data.Install.RootDevice == "" {
+		return fmt.Errorf("%s", "install.rootdevice is required")
+	}
+
+	if data.Install.DataDevice == "" {
+		data.Install.Device = data.Install.RootDevice
+	}
+
+	// Should probably have a canonical location to fetch rootfs - github?/s3?
```

Comment:
Agree. Thinking of hosting in S3. They allow custom domains, so something like `downloads.autonomy.io` could be backed by S3. On the other hand GitHub is probably simpler. GitHub might be a good first start, but that would mean that only releases with attached files could be downloaded. With S3 we could push all artifacts from every build. Something to talk about. Perhaps even old school static Nginx running in AWS somewhere.